### PR TITLE
Consolidate the warning stacklevel in KubernetesPodTrigger

### LINKED ISSUE #34424: Consolidate the warning stacklevel in KubernetesPodTrigger
### Body

Use stacklevel 2 instead of the default 1:
https://github.com/apache/airflow/blob/1b122c15030e99cef9d4ff26d3781a7a9d6949bc/airflow/providers/cncf/kubernetes/triggers/pod.py#L103-L106

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by anukrati1507**
Used stacklevel 2 instead of the default 1 in 
Closes: #34424 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by amoghrajesh — APPROVED** [review]
Changes look good to me +1. Let us wait for the CI
@hussein-awala WDYT?

File: airflow/providers/cncf/kubernetes/triggers/pod.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ def __init__(
             warnings.warn(
                 "`should_delete_pod` parameter is deprecated, please use `on_finish_action`",
                 AirflowProviderDeprecationWarning,
+                stacklevel=2
```
Comment (amoghrajesh) [inline airflow/providers/cncf/kubernetes/triggers/pod.py]: On second thought, can you add category=AirflowProviderDeprecationWarning?

File: airflow/providers/cncf/kubernetes/triggers/pod.py [hunk 2]
```diff
@@ -102,7 +102,8 @@ def __init__(
         if should_delete_pod is not None:
             warnings.warn(
                 "`should_delete_pod` parameter is deprecated, please use `on_finish_action`",
-                AirflowProviderDeprecationWarning,
+                category=AirflowProviderDeprecationWarning,
+                stacklevel=2
```
Comment (hussein-awala) [inline airflow/providers/cncf/kubernetes/triggers/pod.py]:
```suggestion
                stacklevel=2,
```

**Review by hussein-awala — APPROVED** [review]
LGTM

I recommend reading https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst for the next contributions.

**Review by uranusjr — APPROVED** [review]

**Pull request merged**